Source/Orts.Simulation/Simulation/Physics/Train.cs

```diff
@@ -281,10 +281,20 @@ public float CheckDistanceM
                 float sectionStart = -offset;
 
                 // Get all track circuit sections in front of the train
-                List<int> sectionIndexes = signalRef.ScanRoute(this, position.TCSectionIndex, position.TCOffset,
-                        position.TCDirection, true, -1, true, false,
-                        false, false, true, false, false, false, false, IsFreight);
-                List<TrackCircuitSection> sections = new List<TrackCircuitSection>();
+                //List<int> sectionIndexes = signalRef.ScanRoute(this, position.TCSectionIndex, position.TCOffset,
+                //        position.TCDirection, true, -1, true, false,
+                //        false, false, true, false, false, false, false, IsFreight);
+                //List<TrackCircuitSection> sections = new List<TrackCircuitSection>();
+
+                var sectionIndexes = new List<int>();
```

Comment:
In this line one additional new list is created even if the error doesn't occure. The proper method would be just to declare here and not assign.

Reply:
Hi Peter,

Thanks for looking at this. You are quite correct - that an empty list is created and is then not used.

It's a practice that my team has adopted so that we never have to check for lists which are null. Instead they are initially empty.

The developers do less work in coding and debugging and the computer does a bit more.

Best wishes,

Chris

Reply:
Chris, the new empty list can be created in the "else" section to avoid the null-check. But creating 1 additional new list when the error doesn't occur (probably 99% of the cases) just spins the garbage collector, which in a video game causes micro-stuttering. Creating new non-elementary objects in the Update() and Render() loops must be avoided as much as possible.